### PR TITLE
Add SortedSet

### PR DESCRIPTION
**Pull request by bluetrain**
- Ruby 3.x removes Sorted Set. So, copy the SortedSet implementation from 2.7.x.

### CODE REVIEW

File: src/testup/report/SortedSet.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      end
+    end
+  end
+
+  def initialize(*args, &block) # :nodoc:
+    SortedSet.setup
+    @keys = nil
+    super
+  end
+end
```
Comment (thomthom) [inline src/testup/report/SortedSet.rb:149]: We need to account for TestUp also running on older SketchUp versions where `set` do include `SortedSet`:

```
class SortedSet < Set
# ...
end unless defined?(SortedSet)
```

File: src/testup/report/SortedSet.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+# Copyright (c) 2002-2016 Akinori MUSHA <knu@iDaemons.org>
+#
+# Documentation by Akinori MUSHA and Gavin Sinclair.
+#
+# All rights reserved.  You can redistribute and/or modify it under the same
+# terms as Ruby.
+#
+#-------------------------------------------------------------------------------
+
+require 'set'
```
Comment (thomthom) [inline src/testup/report/SortedSet.rb:12]: I'd add a comment here that explains why we have this file here.

File: src/testup/report/collection.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #
 #-------------------------------------------------------------------------------
 
-require 'set'
+require 'testup/report/SortedSet'
```
Comment (thomthom) [inline src/testup/report/collection.rb:8]: I don't think `SortedSet.rb` should be under the `report` folder. Instead I'd move it up a level to be at `testup/sorted_setrb` (Also adjusting file name to better match existing files)

**Review by thomthom — CHANGES_REQUESTED** [review]
- [ ] Need a guard against defining `SortedSet` under Ruby 2.x
- [ ] A comment to explain the purpose of the new file.
- [ ] Relocate the file up one level

File: src/testup/report/SortedSet.rb (hunk 1 above)
Comment (bluetrain) [reply src/testup/report/SortedSet.rb:149]: I tested with Ruby 2.7. I can test with earlier versions. Unless there was a break in compatibility, I don't expect any issues.

Comment (bluetrain) [reply src/testup/report/SortedSet.rb:149]: It seems like SortedSet is defined as a constant in Ruby 3.x:
puts RUBY_VERSION
3.1.1
=> nil
defined?(SortedSet)
=> "constant"

Comment (thomthom) [reply src/testup/report/SortedSet.rb]: Yea, as long as all Ruby version can use this version then we're good.

**Review by thomthom — APPROVED** [review]
- [ ] Squash

**Pull request merged**